core/src/main/java/com/microsoft/applicationinsights/channel/concrete/localforwarder/LocalForwarderTelemetriesTransmitter.java

```diff
@@ -0,0 +1,150 @@
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+public class LocalForwarderTelemetriesTransmitter implements TelemetriesTransmitter<Telemetry> {
+
+    private ManagedChannel channel;
+    private AITelemetryServiceStub asyncService;
+
+    private final ScheduledExecutorService executor;
+    private final ExecutorService grpcServiceExecutor;
+
+    private StreamObserver<AiResponse> responseObserver = new StreamObserver<AiResponse>() {
+        private final LocalForwarderTelemetriesTransmitter thiz = LocalForwarderTelemetriesTransmitter.this;
+        @Override
+        public void onNext(AiResponse value) {
+            trace("Response received: %s", value.toString());
+        }
+
+        @Override
+        public void onError(Throwable t) {
+            error("Error encountered:%n%s", ExceptionUtils.getStackTrace(t));
+        }
+
+        @Override
+        public void onCompleted() {
+            trace("Send completed.");
+        }
+    };
+
+    @VisibleForTesting
+    LocalForwarderTelemetriesTransmitter(ManagedChannelBuilder channelBuilder, boolean createDefaultGrpcExecutor, int instanceId) {
+        Preconditions.checkNotNull(channelBuilder, "channelBuilder");
+
+        if (createDefaultGrpcExecutor) {
+            this.grpcServiceExecutor = new ThreadPoolExecutor(1, 10, 30, TimeUnit.SECONDS, new LinkedBlockingDeque<Runnable>(),
+                    ThreadPoolUtils.createDaemonThreadFactory(LocalForwarderTelemetriesTransmitter.class, String.format("%d-grpcPool", instanceId)));
+            channelBuilder.executor(this.grpcServiceExecutor);
+        } else {
+            this.grpcServiceExecutor = null;
+        }
+        this.executor = Executors.newScheduledThreadPool(1, ThreadPoolUtils.createDaemonThreadFactory(LocalForwarderTelemetriesTransmitter.class, instanceId));
+        this.channel = channelBuilder.build();
+        this.asyncService = AITelemetryServiceGrpc.newStub(channel);
+    }
+
+    @VisibleForTesting
+    void setResponseObserver(StreamObserver<AiResponse> responseObserver) {
+        this.responseObserver = responseObserver;
+    }
+
+    @Override
+    public boolean scheduleSend(final TelemetriesFetcher<Telemetry> telemetriesFetcher, long value, TimeUnit timeUnit) {
+        final Collection<Telemetry> telemetries = telemetriesFetcher.fetch();
+        try {
+            this.executor.schedule(getSenderRunnable(telemetries), value, timeUnit);
+            return true;
+        } catch (Exception e) {
+            error("Error in scheduledSend. %d items not sent", telemetries.size());
+        }
+        return false;
+    }
+
+    private Runnable getSenderRunnable(final Collection<Telemetry> telemetries) {
+        return new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    StreamObserver<TelemetryBatch> requestObserver = asyncService.sendTelemetryBatch(responseObserver);

```

Comment:
@littleaj please note this is not a blocking comment. I was looking deeper into gRPC and how the channel could react in high scale applications. One problem that can be pronounced is of [BackPressure](https://dzone.com/articles/applying-back-pressure-when). Generally when an observable is kind of "Hot Observable" and keeps emitting events at it's own pace - there can be situations when your observer is unable to consume it at a pace at which the observable is emitting it. There are ways to handle this, batching being one. 

I didn't find a good built in implementation for native grpc that we are using here, however the s[alesforce wrapper of Rxjava based gRPC](https://github.com/salesforce/reactive-grpc) provides good built in ways to handle backpressure. May be you could take a look later!